test/local/pool-erc20.test.ts

```diff
@@ -0,0 +1,359 @@
+    sut = (await deployContract(owner, PoolArtifact)) as Pool;
+
+    let fixedRatesCalculator = (await deployContract(owner, FixedRatesCalculatorArtifact, [toWei("0.05"), toWei("0.1")])) as FixedRatesCalculator;
+    let borrowersRegistry = (await deployContract(owner, OpenBorrowersRegistryArtifact)) as OpenBorrowersRegistry;
+
+    await sut.initialize(fixedRatesCalculator.address, borrowersRegistry.address, ZERO, ZERO);
+  });
+
+  describe("transfer", () => {
+
+    it("should revert if not enough balance", async () => {
+      await sut.connect(user1).deposit({value: toWei("1.0")});
+
+      await expect(sut.connect(user1).transfer(user2.address, toWei("1.1")))
+        .to.be.revertedWith("ERC20: transfer amount exceeds balance");
+    });
+
+    it("should accumulate user5 interests prior transferring the funds", async () => {
+      // given
+      await sut.connect(user1).deposit({value: toWei("1.0")});
+      await time.increase(time.duration.years(1));
+
+      // note: after accumulating interests user5 should be able to transfer
+      // more funds than originally deposited
+      // when
+      await sut.connect(user1).transfer(user2.address, toWei("1.04"));
+
+      // then
+      expect(await balanceOf(user2)).to.be.equal(1.04);
+      expect(await balanceOf(user1)).to.be.closeTo(0.011271, 0.000001);
+    });
+
+    it("should accumulate user2 interests prior transferring the funds", async () => {
+      // given
+      await sut.connect(user1).deposit({value: toWei("1.0")});
+      await sut.connect(user2).deposit({value: toWei("2.0")});
+      await time.increase(time.duration.years(1));
+
+      // when
+      await sut.connect(user1).transfer(user2.address, toWei("1.05"));
+
+      // then
+      expect(await balanceOf(user1)).to.be.closeTo(0.001271, 0.000001);
+      expect(await balanceOf(user2)).to.be.closeTo(3.152542, 0.000001);
+    });
+
+  });
+
+  describe("approve", () => {
+    it("should revert if value to approve higher than current balance", async () => {
+      await sut.connect(user1).deposit({value: toWei("1.0")});
+
+      await expect(sut.connect(user1).approve(user2.address, toWei("1.01")))
+        .to.be.revertedWith("ERC20: approve amount exceeds balance");
+
+    });
+
+    it("should revert if value to approve higher than current balance (after accumulating interest)", async () => {
+      await sut.connect(user1).deposit({value: toWei("1.0")});
+      await time.increase(time.duration.years(1));
+
+      await expect(sut.connect(user1).approve(user2.address, toWei("1.06")))
+        .to.be.revertedWith("ERC20: approve amount exceeds balance");
+    });
+
+    it("should properly accumulate interest rate before approving", async () => {
+      // given
+      await sut.connect(user1).deposit({value: toWei("1.0")});
+      await time.increase(time.duration.years(1));
+
+      // when
+      // note: even though only 1.0 was deposited, 1.05 can be approved
+      // thanks to interests being accumulated
```

Comment:
Correct me if I'm wrong, but I believe that interest(s) in the financial context of received money for providing something is an uncountable noun. There are many places in the code so I will just leave this last comment but it addresses all of the other places as well.

Normally I would not care too much but in this context it is a 'false-friend' and changes the meaning of the word :)

Reply:
Seems that you are right. From a short research 99% of times it is used as uncountable in the financial context

Reply:
Replaced 'interests' with 'interest'